### PR TITLE
Bug fix/8334 duplicate names are getting created on updating elements

### PR DESCRIPTION
**Pull request by ravirk91**
Bug 8334:Duplicate Names are getting created on Updating Elements

### CODE REVIEW

File: Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs [hunk 1]
```diff
@@ -386,19 +386,72 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
...
+                        }
+
+                    }
+
+                    if (toRemoveAddedFoundItem != null)
+                    {
+                        addedElements.Remove(toRemoveAddedFoundItem);
+                    }
+
+                    if (matchingElementFound==false)
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs]: Codacy Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4592337)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Ginger/Ginger/pullRequest?prid=4592337&bid=15419331)

File: Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs [hunk 2]
```diff
@@ -386,19 +386,72 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
...
+                            toRemoveAddedFoundItem = newElement;
+                            //element found and updated, so exit from loop
+                            break;
+                        }
+
+                    }
+
+                    if (toRemoveAddedFoundItem != null)
+                    {
+                        addedElements.Remove(toRemoveAddedFoundItem);
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs]: why removing again if already removed on line 424

File: Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs [hunk 3]
```diff
@@ -386,19 +386,72 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
...
+                            DeltaViewElements.Add(ConvertElementToDelta(unidentifiedElement, eDeltaStatus.Changed, unidentifiedElement.ElementGroup, true, "Element Property Updated"));
+                            
+                            var itemToRemove = DeltaViewElements.Where(x => x.ElementInfo.Guid == newElement.ElementInfo.Guid).FirstOrDefault();
+                            if (itemToRemove != null)
+                            {
+                                DeltaViewElements.Remove(itemToRemove);
+                            }
+
+                            // add found newElment in removeitem 
+                            toRemoveAddedFoundItem = newElement;
```
Comment (JinendraGhodke) [inline Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs]: is it needed?

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4592337&bid=15419331)

File: Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs [hunk 4]
```diff
@@ -386,19 +386,90 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
...
+                                        existingProperty.DeltaExtraDetails = "Property changed";
+                                    }
+                                }
+                            }
+                            matchingElementFound = true;
+
+                            var mathchedItemIndex = DeltaViewElements.IndexOf(DeltaViewElements.Where(x => x.ElementInfo.Guid.Equals(newElement.ElementInfo.Guid)).FirstOrDefault());
+                            var item = ConvertElementToDelta(deletedElement, eDeltaStatus.Changed, deletedElement.ElementGroup, false, "Property Changed");
+                            item.Properties = deltaControlProp;
+                            if (mathchedItemIndex != -1)
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4592337)

**Comment by ambarkhan** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4592337&bid=15419331)

File: Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs [hunk 5]
```diff
@@ -386,19 +386,93 @@ private void SetMatchingElementDeltaDetails(ElementInfo existingElement, Element
...
+                            }
+                            matchingElementFound = true;
+
+                            var mathchedItemIndex = DeltaViewElements.IndexOf(DeltaViewElements.Where(x => x.ElementInfo.Guid.Equals(newElement.ElementInfo.Guid)).FirstOrDefault());
+                            //update path of element
+                            deletedElement.Path = newElement.ElementInfo.Path;
+
+                            var item = ConvertElementToDelta(deletedElement, eDeltaStatus.Changed, deletedElement.ElementGroup, true, "Property Changed");
+                            item.Properties = deltaControlProp;
+                            if (mathchedItemIndex != -1)
```
Comment (ambarkhan) [inline Ginger/GingerCoreNET/Application Models/Delta/PomDelta/PomDeltaUtils.cs]: Codacy Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4592337)

**Pull request merged**